File: MicroCRM/WebApp/src/models/order.ts
import { Customer } from "./customer";
import { OrderLine } from "./order-line";

/**
 * The order model class.
 */
export class Order {
  public id: string;
  public orderStatus: string;
  public orderNumber: string;
  public customer: Customer;
  public date: string;
  public orderLines: OrderLine[];
  public total: number;
}

File: src/backend/MicroCRM/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using MicroCRM.Entities.Abstract;

namespace MicroCRM.Entities
{
    [Table("Orders")]
    public class Order : LinkedEntity
    {
        #region Public properties

        public OrderStatus OrderStatus { get; set; } = OrderStatus.Created;

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int OrderNumber { get; set; }
        
        [ForeignKey("CustomerId")]
        public Guid CustomerId { get; set; }

        public virtual Customer Customer { get; set; }

        public virtual IEnumerable<OrderLine> OrderLines { get; set; }

        #endregion
    }
}
